pkg/ui/src/redux/customAnalytics/customAnalyticsSagas.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+import { call, put, takeEvery } from "redux-saga/effects";
+import Analytics from "analytics-node";
+
+import { PayloadAction } from "src/interfaces/action";
+import { COCKROACHLABS_ADDR } from "src/util/cockroachlabsAPI";
+import {
+  completeEmailSubscriptionSignUp,
+  EMAIL_SUBSCRIPTION_SIGN_UP,
+  EmailSubscriptionSignUpPayload,
+} from "./customAnanlyticsActions";
+
+export type AnalyticsClientTarget = "email_sign_up";
+
+// TODO (koorosh): has to be moved out from code base
+const EMAIL_SIGN_UP_CLIENT_KEY = "72EEC0nqQKfoLWq0ZcGoTkJFIG9G9SII";
```

Comment:
For now, we'll just leave this in-line since we do the same in `analytics.ts`, but good to leave the TODO.